typewise-alert.c

```diff
@@ -1,71 +1,79 @@
-#include "typewise-alert.h"
 #include <stdio.h>
+#include "typewise-alert.h"
+#include"checker.h"
+
+BreachType inferBreach(double value, double lowerLimit, double upperLimit)
+{
+	BreachType breach = NORMAL;
+	checkLowLimitBreach(value, lowerLimit, &breach);
+	checkUpLimitBreach(value, upperLimit, &breach);
+	return breach;
+}
+
+BreachType classifyTemperatureBreach(CoolingType coolingType, double temperatureInC) 
+{
+  	int lowerLimit = 0;
+  	int upperLimit = 0;
 
-BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
-  if(value < lowerLimit) {
-    return TOO_LOW;
-  }
-  if(value > upperLimit) {
-    return TOO_HIGH;
-  }
-  return NORMAL;
+  	evaluateLimit_PASSIVE_COOLING(coolingType, &lowerLimit, &upperLimit);
```

Comment:
All three methods are called. It can be improved by use of an efficient data structure (dictionary of coolingtype vs limits ) and single method call.